Makefile

```diff
@@ -88,9 +86,28 @@ test-unit: ## Run the unit tests
 verify: manifests generate format
 	git diff --exit-code
 
+###
 # Utilities.
-.PHONY: controller-gen
-
-controller-gen: vendor ## Find or download controller-gen 
-CONTROLLER_GEN=$(Q)go run -mod=vendor ./vendor/sigs.k8s.io/controller-tools/cmd/controller-gen
-
+###
+
+# go-get-tool will 'go get' any package $2 and install it to $1.
+PROJECT_DIR := $(shell dirname $(abspath $(lastword $(MAKEFILE_LIST))))
+define go-get-tool
+@[ -f $(1) ] || { \
+set -e ;\
+TMP_DIR=$$(mktemp -d) ;\
+cd $$TMP_DIR ;\
+go mod init tmp ;\
+echo "Downloading $(2)" ;\
+GOBIN=$(PROJECT_DIR)/bin go get $(2) ;\
```

Comment:
Just as a heads up.. this is fine for now, but it's going to break when we bump to Go 1.18. See https://github.com/operator-framework/operator-sdk/issues/5481

Reply:
That's a good call out - I'm personally not super worried if we're committed to tracking this here as an issue (or ideally in jira ticket) as it looks like a fairly straight forward change, and removing the vendor directory is a step in the right directory (both in the context of upstream, and especially downstream) imo.

Reply:
I'll create an ticket in this repository once this PR merges to track this.